org.uqbar.project.wollok.lib/src/wollok/lang/WDictionary.xtend

```diff
@@ -0,0 +1,60 @@
+package wollok.lang
+
+import java.util.Collection
+import java.util.Map
+import java.util.Map.Entry
+import org.eclipse.xtend.lib.annotations.Accessors
+import org.uqbar.project.wollok.interpreter.api.WollokInterpreterAccess
+import org.uqbar.project.wollok.interpreter.core.WollokObject
+import org.uqbar.project.wollok.interpreter.nativeobj.JavaWrapper
+
+import static extension org.uqbar.project.wollok.interpreter.nativeobj.WollokJavaConversions.*
+import static extension org.uqbar.project.wollok.lib.WollokSDKExtensions.*
+
+class WDictionary implements JavaWrapper<Map> {
+	
+	val WollokObject wollokInstance
+	@Accessors var Map wrapped
+	protected extension WollokInterpreterAccess = new WollokInterpreterAccess
+		 
+	new(WollokObject o) {
+		wollokInstance = o
+		wrapped = newHashMap
+	}
+	
+	def clear() { wrapped.clear }
+	
+	def basicGet(WollokObject key) { wrapped.get(getInternalKey(key)) }
+	
+	def put(WollokObject key, WollokObject value) { wrapped.put(key, value) }
+
+	def void remove(WollokObject key) {
+		wrapped.remove(getInternalKey(key))
+	}
+	
+	private def Object getInternalKey(WollokObject key) {
```

Comment:
Even when we are not performance maniacs... I think this breaks the idea of having a dictionary, we shouldn't iterate through the keys to find the right object. ¿Is it too difficult to have a hash+equals which are compliant with the needs of the hashmap?